appserver/tests/payara-samples/samples/remote-ejb-tracing/remote-ejb-tracing-client/src/test/java/fish/payara/samples/remote/ejb/tracing/RemoteEjbClientIT.java

```diff
@@ -0,0 +1,106 @@
+ */
+package fish.payara.samples.remote.ejb.tracing;
+
+import io.opentracing.Scope;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+import org.junit.Assert;
+import org.junit.Test;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import java.util.Properties;
+
+public class RemoteEjbClientIT {
+
+    @Test
+    public void executeRemoteEjbMethodIT() {
+        Properties contextProperties = new Properties();
+        contextProperties.setProperty(Context.INITIAL_CONTEXT_FACTORY, "com.sun.enterprise.naming.SerialInitContextFactory");
+        contextProperties.setProperty("org.omg.CORBA.ORBInitialHost", "localhost");
+        contextProperties.setProperty("org.omg.CORBA.ORBInitialPort", "3700");
+
+        try {
+            Context context = new InitialContext(contextProperties);
+            EjbRemote ejb = (EjbRemote) context.lookup("java:global/remote-ejb-tracing-server/Ejb");
+
+            Tracer tracer = GlobalTracer.get();
+
+            try (Scope scope = tracer.buildSpan("ExecuteEjb").startActive(true)) {
+                Span span = scope.span();
+                span.setBaggageItem("Wibbles", "Wobbles");
+                String baggageItems = ejb.annotatedMethod();
+                Assert.assertTrue("Baggage items didn't match, received: " + baggageItems,
+                        baggageItems.equals("\nWibbles : Wobbles\n"));
+
+                span.setBaggageItem("Nibbles", "Nobbles");
+                baggageItems = ejb.nonAnnotatedMethod();
+                Assert.assertTrue("Baggage items didn't match, received: " + baggageItems,
+                        baggageItems.contains("Wibbles : Wobbles"));
+                Assert.assertTrue("Baggage items didn't match, received: " + baggageItems,
+                        baggageItems.contains("Nibbles : Nobbles"));
+
+                span.setBaggageItem("Bibbles", "Bobbles");
+                baggageItems = ejb.shouldNotBeTraced();
+                Assert.assertTrue("Baggage items didn't match, received: " + baggageItems,
+                        baggageItems.contains("Wibbles : Wobbles"));
+                Assert.assertTrue("Baggage items didn't match, received: " + baggageItems,
+                        baggageItems.contains("Nibbles : Nobbles"));
+                Assert.assertTrue("Baggage items didn't match, received: " + baggageItems,
+                        baggageItems.contains("Bibbles : Bobbles"));
+
+                baggageItems = ejb.editBaggageItems();
+                Assert.assertTrue("Baggage items didn't match, received: " + baggageItems,
+                        baggageItems.contains("Wibbles : Wabbles"));
+                Assert.assertTrue("Baggage items didn't match, received: " + baggageItems,
+                        baggageItems.contains("Nibbles : Nabbles"));
+                Assert.assertTrue("Baggage items didn't match, received: " + baggageItems,
```

Comment:
Some duplication going on...

Reply:
Better use MatcherAssert.assertThat("Baggage items " + baggageItems, baggageItems, hasItems(....));

Reply:
And traditional convention in tests: use static imports for all assert* methods.